components/screens/ShoppingCart.js

```diff
@@ -1,13 +1,46 @@
-import React from 'react';
+import React, {useState} from 'react';
+import { Button } from 'react-native';
 import { View, ScrollView, Text, StyleSheet } from 'react-native';
 import NavBar from '../NavBar';
 
-function ShoppingCart({navigation}) {
+function ShoppingCart({ screenProps,navigation }) {
+
+    
+    const createList = () => {
+        return screenProps.shoppingCart.map((element) => {
+            return (
+                <View style={{margin:10}}>
+                    <Text>{element.name}</Text>
+                    <Text>{element.price}</Text>
+                    <Button title='remove' onPress={()=>removeShoppingCartItem(element.key)}></Button>
+                </View>
+            );
+        });
+    };
+
+    const [list, setList] = useState(createList);
+
+    const removeShoppingCartItem = (itemKey) =>{
+
+         let shoppingCart = screenProps.shoppingCart;
+
+        for(let index = 0; index < shoppingCart.length; index++){
+            if(shoppingCart[index].key == itemKey){
+                shoppingCart.splice(index,1);
+                break;
+            }
+        }
+
```

Comment:
Is it ok to use splice() here? I know that its directly mutating the screenProps.shoppingCart but I want this function to change that array anyways. Otherwise I'm kind of stuck because I can't find a way to get the dom to update.